GVFS/GVFS/CommandLine/StatisticsVerb.cs

```diff
@@ -0,0 +1,209 @@
+
+            this.Output.WriteLine("\nMost hydrated top level directories:");
+
+            int maxDirectoryNameLength = 0;
+            for (int i = 0; i < 5 && i < topLevelDirectoriesByHydration.Count; i++)
+            {
+                maxDirectoryNameLength = Math.Max(maxDirectoryNameLength, topLevelDirectoriesByHydration[i].Length);
+            }
+
+            for (int i = 0; i < DirectoryDisplayCount && i < topLevelDirectoriesByHydration.Count; i++)
+            {
+                string dir = topLevelDirectoriesByHydration[i].PadRight(maxDirectoryNameLength);
+                string percent = this.FormatPercent(enlistmentStatistics.GetHydrationOfDirectory(topLevelDirectoriesByHydration[i]));
+                this.Output.WriteLine(" " + percent + " | " + dir);
+            }
+
+            bool healthyRepo = (enlistmentStatistics.GetPlaceholderPercentage() + enlistmentStatistics.GetModifiedPathsPercentage()) < (0.5);
+
+            this.Output.WriteLine("\nRepository status: " + (healthyRepo ? "Healthy" : "Unhealthy"));
+        }
+
+        /// <summary>
+        /// Takes a fractional double and formats it as a percent taking exactly 4 characters with no decimals
+        /// </summary>
+        /// <param name="percent">Fractional double to format to a percent</param>
+        /// <returns>A 4 character string formatting the percent correctly</returns>
+        private string FormatPercent(double percent)
+        {
+            return percent.ToString("P0").PadLeft(4);
+        }
+
+        /// <summary>
+        /// Parse a line of the git index coming from the ls-tree endpoint in the git process to get the path to that file
+        /// </summary>
+        /// <param name="line">The line from the output of the git index</param>
+        /// <returns>The path extracted from the provided line of the git index</returns>
+        private string TrimGitIndexLine(string line)
+        {
+            return line.Substring(line.IndexOf("blob") + 46);
+        }
+
+        /// <summary>
+        /// Talk to the mount process across the named pipe to get a list of the modified paths
+        /// </summary>
+        /// <param name="enlistment">The enlistment being operated on</param>
+        /// <returns>An array containing all of the modified paths in string format</returns>
+        private string[] GetModifiedPathsFromPipe(GVFSEnlistment enlistment)
+        {
+            using (NamedPipeClient pipeClient = new NamedPipeClient(enlistment.NamedPipeName))
+            {
+                string[] modifiedPathsList = { };
+
+                if (!pipeClient.Connect())
+                {
+                    this.ReportErrorAndExit("Unable to connect to GVFS.  Try running 'gvfs mount'");
+                }
+
+                try
+                {
+                    NamedPipeMessages.Message modifiedPathsMessage = new NamedPipeMessages.Message(NamedPipeMessages.ModifiedPaths.ListRequest, "1");

```

Comment:
Style suggestion:  Prefer named constants for hard coded magic values.

For example, instead of:

```
new NamedPipeMessages.Message(NamedPipeMessages.ModifiedPaths.ListRequest, "1");
```

Have a common spot where "1" is defined and use it here:

*In ModifiedPaths class (see GVFS/GVFS.Common/NamedPipes/NamedPipeMessages.cs)*

Add the following:

```
public const string CurrentVersion = "1";
```

*In InProcessMount*

- Remove `private const string ModifiedPathsVersion = "1";
- Update the code to use the `CurrentVersion` you've added to `ModifiedPaths`

*In this code*

```
new NamedPipeMessages.Message(NamedPipeMessages.ModifiedPaths.ListRequest, NamedPipeMessages.ModifiedPaths.CurrentVersion);
```

Reply:
OK good catch. That makes a lot of sense, I'll try to remember to structure things that way in the future!